tools/formatter/format.mk

```diff
@@ -74,3 +74,21 @@ lint/line-endings:
 		echo -e "found files with CRLF line endings: \n$$_files"; false; \
 	else echo all files have unix line endings; \
 	fi
+
+output/docker_mlperf_formatter.stamp: tools/formatter/Dockerfile
+	docker build --progress=plain \
+		--build-arg UID=`id -u` --build-arg GID=`id -g` \
+		-t mlperf/formatter tools/formatter
+	# need to clean flutter cache first else we will have error when running `dart run import_sorter:main` later in docker
+	cd flutter && flutter clean
```

Comment:
If we need this, we probably need this before each `docker/format` command.
We only build image 1 time, and something can invalidate Flutter cache later, when we already have formatter docker .stamp file.

Reply:
We should not clear cache on every run. In my test, running `flutter clean` only one time there is enough. Subsequent run `make docker/format` or `flutter run` worked fine.

Reply:
Theoretically we shouldn't have to clear Flutter cache. In practice sometimes we do, but it doesn't have anything to do with building docker image.

I personally experience issues that are resolved by `flutter clean` when I switch between platforms. For example, if I use Flutter in Windows, and then launch Ubuntu in WSL and use mounted flutter folder there. Or something like this, I didn't try to find the exact conditions and commands.

If you got errors when running flutter in docker first time, then you can potentially get the same error next time when you switch between local flutter and docker.

Reply:
I didn't try to find out why `flutter clean` is needed the first time when using `make docker/format`. But as stated, I can run `make docker/format` to run flutter in Docker container, or run `flutter run` on my Mac, and switch between 2 without any problem.

Reply:
Ok, mentioning the clean command in the documentation should be enough.

Reply:
If the problem may occur, then it is not worth to clear cache by every run. I added a note about clearing flutter cache in  `CONTRIBUTING.md`.